timemachine/datasets/utils.py

```diff
@@ -1,10 +1,26 @@
 from importlib import resources
-from typing import List
+from typing import List, Optional, Set
 
 from rdkit import Chem
 
+from timemachine.fe.utils import get_mol_name
 
-def fetch_freesolv() -> List[Chem.Mol]:
+
+def fetch_freesolv(n_mols: Optional[int] = None, filter_mols: Optional[Set[str]] = None) -> List[Chem.Mol]:
```

Comment:
Naming preference: `filter_mols` -> `exclude_mols` or similar

("filter" could be misinterpreted as only _keeping_ select mols, rather than only excluding select mols, e.g. based on how python built-in `filter` works)